include/pybind11/pybind11.h

```diff
@@ -887,19 +887,28 @@ class cpp_function : public function {
     }
 };
 
-/// Wrapper for Python extension modules
+/// Wrapper for Python modules
 class module_ : public object {
 public:
     PYBIND11_OBJECT_DEFAULT(module_, object, PyModule_Check)
 
     /// Create a new top-level Python module with the given name and docstring
-    PYBIND11_DEPRECATED("Use PYBIND11_MODULE or module_::create_extension_module instead")
     explicit module_(const char *name, const char *doc = nullptr) {
-#if PY_MAJOR_VERSION >= 3
-        *this = create_extension_module(name, doc, new PyModuleDef());
+#if defined(PYPY_VERSION)
+        m_ptr = PyModule_New(const_cast<char *>(name));
 #else
-        *this = create_extension_module(name, doc, nullptr);
+        m_ptr = PyModule_New(name);
 #endif
+        if (m_ptr == nullptr)
+            pybind11_fail("Internal error in module_::module_()");
```

Comment:
"Internal error" sounds pretty vague, but I see pybind uses it all over the place and nobody complained.

Reply:
Fixed to the almost-as-vague-but-slightly-better-and-more-consistent `"Could not allocate module object!"`. I don't know where I got `"Internal error"`. I suppose it was already there?